### DISCUSSION TITLE
How many apps can be supported by a single deployment

### DISCUSSION THREAD

**Discussion by DanielYWoo** [argoproj/argo-cd#18630] [General]
We currently have 3000+ apps with 30+ clusters and we see some performance issues. What is the recommended volume of workload for a typical argocd deployment? And is there a document about performance tuning for large workload?

**Comment by kotyara85** [comment]
@DanielYWoo What kind of performance issues do you observe?

**Comment by DanielYWoo** [comment]
The main problem is that the homepage UI won't load due to the huge amount of data to load, I have to concatenate query string to the homepage URL to filter out some apps. Secondly, we also have some smaller ArgoCD instances which only manage hundreds of apps, the sync takes much longer in the big instance than other small instances.
